pydantic/main.py

```diff
@@ -978,6 +992,9 @@ def validate_model(  # noqa: C901 (ignore complexity)
             using_name = True
 
         if value is _missing:
+            if config.partial:
```

Comment:
I feel like making the field non required is better and more expected.
If you introspect your models or generate a JSON schema, you expect nothing in required keys

Reply:
That's a good point! I was being overly protective in the event someone didn't pass `exclude_unset=True` to `dict()`. By short-circuiting here, it protects against ever accidentally getting `null` values for non-nullable fields.